sympy/sets/sets.py

```diff
@@ -2516,3 +2485,46 @@ def set_pow(x, y):
 def set_function(f, x):
     from sympy.sets.handlers.functions import _set_function
     return _set_function(f, x)
+
+
+@dispatch(Interval, FiniteSet)
+def _eval_Eq(lhs, rhs): # noqa: F811
+    return false
```

Comment:
Rather than have these things here I think they should go in a module in `sympy/sets/handlers/`